internal/templates/containercapabilities/template.go

```diff
@@ -0,0 +1,218 @@
+package containercapabilities
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/matcher"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/containercapabilities/internal/params"
+	"golang.stackrox.io/kube-linter/internal/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	reservedCapabilitiesAll             = "all"
+	matchLiteralReservedCapabilitiesAll = "(?i)" + reservedCapabilitiesAll
+)
+
+var (
+	literalReservedCapabilitiesAllMatcher func(string) bool
+)
+
+func checkCapabilityDropList(
+	containerName string,
+	paramCapMatchers map[string]func(string) bool,
+	scCaps *v1.Capabilities,
+	forbidAll bool,
+) *diagnostic.Diagnostic {
+	if forbidAll {
+		// Specified to drop all. Instead of trying to find a match to this paramCap, we should
+		// make sure the special ReservedCapabilitiesAll is found in the DROP list as well
+		for _, scCap := range scCaps.Drop {
+			if literalReservedCapabilitiesAllMatcher(string(scCap)) {
+				// Found "all" in drop list as well
+				return nil
+			}
+		}
+		return &diagnostic.Diagnostic{
+			Message: fmt.Sprintf(
+				"container %q has DROP capabilities: %q, but in fact all capabilities are required to be dropeed",
+				containerName,
+				scCaps.Drop),
+		}
+	}
+
+	// Every forbidden capability specified by param should be found in the DROP list
+	for paramCap, paramCapMatcher := range paramCapMatchers {
+		var found bool
+		for _, scCap := range scCaps.Drop {
+			// User can specify to drop "all" under containers as well
+			if paramCapMatcher(string(scCap)) || literalReservedCapabilitiesAllMatcher(string(scCap)) {
```

Comment:
In general, I think we should just compute the effective list of capabilities by looking at the "add" and "drop", and pass those into this function.

Reply:
yeah I thought about it as well, but that requires keeping a base list that is consistent with the default caps, which I was not sure if it changes or not.

The cleanest way of doing this would be:
- 1: find a base list
- 2: perform add and drop
- 3: validate against the user specified params.

However, since we don't have that base list, and not sure how safe it is to just hard code one (not only this can changes, this has to also work with different versions of kubernetes server), I instead chose to check it by making sure unwanted caps are:
- 1: not found in ADD list since this overwrites even if it is included in DROP list
- 2: dropped in DROP list